src/database/models/actions.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+module.exports = (sequelize, DataTypes) => {
+	const Action = sequelize.define(
+		'Action',
+		{
+			id: {
+				allowNull: false,
+				autoIncrement: true,
+				primaryKey: true,
+				type: DataTypes.INTEGER,
+			},
+			code: {
+				allowNull: false,
+				unique: true,
+				type: DataTypes.STRING,
+			},
+			description: {
+				type: DataTypes.STRING,
+			},
+			status: {
+				allowNull: false,
+				type: DataTypes.ENUM('ACTIVE', 'INACTIVE'),
+				defaultValue: 'ACTIVE',
+			},
+		},
+		{
+			modelName: 'Action',
+			tableName: 'actions',
+			freezeTableName: true,
+			paranoid: true,
+		}
+	)
+
+	Action.addHook('beforeDestroy', async (instance, options) => {
+		try {
+			// Soft-delete associated Activities records with matching actions
+			await sequelize.models.Activity.update(
```

Comment:
@priyanka-TL We will not delete any activity if user action is deleted. In action delete API check if there is any activity for that action, if yes then don't allow to delete that action.